### ISSUE TITLE
Webmin don't show all active IPv4 addresses

### ISSUE THREAD

**Issue by macmon** [webmin/webmin#1217]
On Webmin the table of all active networks don't show all active IPv4 addresses. I configured this list of (test) IP addresses to the interface ens160:
- 10.0.0.0
- 10.1.0.0
- 10.2.0.0
- 10.3.0.0
- 10.4.0.0
- 192.168.2.3
- 192.168.2.16
- 192.200.0.0

The Webmin GUI don't display all addresses:
![image](https://user-images.githubusercontent.com/7994753/76838759-abe94100-6834-11ea-86e6-3057a832bfb3.png)

The output of command `ip a` is (without lo):
```
2: ens160: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:0c:29:f6:3b:9d brd ff:ff:ff:ff:ff:ff
    inet 192.168.2.16/20 brd 192.168.255.255 scope global dynamic ens160
       valid_lft 519646sec preferred_lft 519646sec
    inet 10.0.0.0/24 scope global ens160
       valid_lft forever preferred_lft forever
    inet 10.1.0.0/24 scope global ens160
       valid_lft forever preferred_lft forever
    inet 10.2.0.0/24 brd 10.2.0.255 scope global ens160:1
       valid_lft forever preferred_lft forever
    inet 10.3.0.0/24 brd 10.3.0.255 scope global ens160:100
       valid_lft forever preferred_lft forever
    inet 192.200.0.0/24 brd 192.200.0.255 scope global ens160:2
       valid_lft forever preferred_lft forever
    inet 10.4.0.0/24 brd 10.4.0.255 scope global ens160:test
       valid_lft forever preferred_lft forever
    inet 192.168.2.3/20 brd 192.168.255.255 scope global secondary ens160:resIP
       valid_lft forever preferred_lft forever
    inet6 fe80::20c:29ff:fef6:3b9d/64 scope link 
       valid_lft forever preferred_lft forever
```

The output of command `ifconfig -a` is (without lo):
```
ens160: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.16  netmask 255.255.240.0  broadcast 192.168.255.255
        inet6 fe80::20c:29ff:fef6:3b9d  prefixlen 64  scopeid 0x20<link>
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)
        RX packets 256500  bytes 32078628 (30.5 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 226021  bytes 45095823 (43.0 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens160:1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.2.0.0  netmask 255.255.255.0  broadcast 10.2.0.255
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)

ens160:2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.200.0.0  netmask 255.255.255.0  broadcast 192.200.0.255
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)

ens160:100: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.3.0.0  netmask 255.255.255.0  broadcast 10.3.0.255
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)

ens160:resIP: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.3  netmask 255.255.240.0  broadcast 192.168.255.255
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)

ens160:test: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.4.0.0  netmask 255.255.255.0  broadcast 10.4.0.255
        ether 00:0c:29:f6:3b:9d  txqueuelen 1000  (Ethernet)
```

I added the IPs with different commands e.g.:
```
ip addr add 10.1.0.0/24 dev ens160
ifconfig ens160:test 10.4.0.0/24
ifconfig ens160:2 192.200.0.0/24
```

The operating system is a Debian 10.2 (Buster).

**Comment by iliarostovtsev** [comment]
Hi,

What Webmin version do you use? If you don't have Webmin 1.942 yet, [grab it here](http://download.webmin.com/devel/deb/webmin_1.942_all.deb) and try again.

**Comment by macmon** [comment]
Sorry, I tested with 1.941. I test with 1.942 again ...

**Comment by macmon** [comment]
I tested with master Branch and only one more IP addres is displayed:
![image](https://user-images.githubusercontent.com/7994753/76863878-7fe3b500-6860-11ea-92aa-9b45fc16d99b.png)

**Comment by iliarostovtsev** [comment]
What is the output of your `/etc/network/interfaces` file?

**Comment by macmon** [comment]
Only the "main" interface is configured (DHCP). The other IPs are added dynamic by some command `ifconfig` or `ip` (see my issue description). The boot configuration is fine. Only the active IPs are not all display.

```# This file describes the network interfaces available on your system
# and how to activate them. For more information, see interfaces(5).

source /etc/network/interfaces.d/*

# The loopback network interface
auto lo ens160
iface lo inet loopback

# ens160
iface ens160 inet dhcp
```

**Comment by iliarostovtsev** [comment]
> Only the "main" interface is configured (DHCP). 

This may be an issue. Could you configure your main interface as _static_, and see if that's still an issue?

**Comment by sutidor** [comment]
I had the same issue, solved it by flipping dynamic and noprefixroute in my case like so:
`elsif ($l =~ /\sinet\s+([0-9\.]+)\/(\d+)(\s+brd\s+(\S+))?\s+scope\s+global(\s+dynamic)?(\s+noprefixroute)?\s+(\S+)/ && $7 eq $ifc{'name'}) {
    # Line like :
    # inet 192.168.178.58/24 brd 192.168.178.255 scope global dynamic noprefixroute eth0
    $ifc{'address'} = $1;
    $ifc{'netmask'} = &prefix_to_mask("$2");
}`
into file net/linux-lib.pl

Basically what you can do if you find a different string is find out what the line says in terminal `ip addr`, then find the line which is most similar, then copy the regex part to regex101.com and fix it:
**/\sinet\s+([0-9\.]+)\/(\d+)(\s+brd\s+(\S+))?\s+scope\s+global(\s+dynamic)?(\s+noprefixroute)?\s+(\S+)/**

**Comment by macmon** [comment]
I configure one of the IP static (192.168.2.16), but the result is the same.

**Comment by jcameron** [comment]
The next release of Webmin will fix up that dynamic / noprefixroute order
